tools/azsdk-cli/README.md

```diff
@@ -1,56 +1,138 @@
+1. Clone the repository:
+    ```sh
+    git clone https://github.com/Azure/azure-sdk-tools.git
+    cd azure-sdk-tools/tools/azsdk-cli
+    ```
 
-### Individual MCP servers that are **referenced** from `hub`
+2. Restore dependencies:
+    ```sh
+    dotnet restore
+    ```
 
-Everything server csproj:
+### Build
 
-```xml
-  <ItemGroup>
-    <PackageReference Include="Azure.SDK.Tools.MCP.ToolName1" Version="1.0.0" />
-    <PackageReference Include="Azure.SDK.Tools.MCP.ToolName2" Version="1.0.0" />
-    <PackageReference Include="Azure.SDK.Tools.MCP.ToolName3" Version="1.0.0" />
+To build the project:
+
+```sh
+dotnet build
+```
+
+### Run
+
+To run the CLI locally:
+
+```sh
+dotnet run --project Azure.Sdk.Tools.Cli -- --help
+```
+
+### Test
+
+To run the tests:
+
+```sh
+dotnet test
 ```
 
-My vision here is that these individual tools should be:
-- implemented as standalone servers
-- when DI-ed into our `hub` server, provide a `route` that can be provided by the hub server
+## Project Structure and Information
 
-This way, users can add local `mcp`
+This project is both a [System.CommandLine](https://learn.microsoft.com/en-us/dotnet/standard/commandline/) app and an [MCP server](https://github.com/modelcontextprotocol/csharp-sdk).
 
-|Pro/Con|Description|
-|---|---|
-|✅|Tools will work standalone for easy development and testing.|
-|✅|Much more granular control over what versions of which tools are present in the hub.|
-|🟥|Users will have to match a server convention at the very least, and implement an interface that makes their actual workload pluggable into the `hub`.|
-|🟥|EngSys will need to add good integration testing of the contract, so that local development -> deployment in `everything server` will never surprise them.|
+### Directory Structure
 
-## Additional features
+- *Azure.Sdk.Tools.Cli* - Core project for cli/mcp logic
+    - *Commands* - Shared classes for CLI commands
+    - *Configuration* - Constants and other classes
+    - *Helpers* - Helper logic for parsing data
+    - *Models* - Shared models, response classes and schemas
+    - *Services* - Helper classes for working with upstream services, e.g. azure, devops, github
+    - *Tools* - CLI commands and MCP tool implementations
+- *Azure.Sdk.Tools.Cli.Tests* - Test project
+- *Azure.Sdk.Tools.Cli.Contract* - Common classes/interfaces
+- *Azure.Sdk.Tools.Cli.Analyzer* - Compilation addons to enforce conventions
+    - Enforce all tools handled by try/catch
+    - Enforce tool inclusion in service registration for dependency injector
 
-Concept of `chaining` of various MCP servers. Can this hub support taking the output of one spoke and shoving it to another? We probably should right?
+### Pipelines
 
+Public CI - https://dev.azure.com/azure-sdk/public/_build?definitionId=7677
 
-## Some random DOs and DON'Ts of this server
+Release - https://dev.azure.com/azure-sdk/internal/_build?definitionId=7684 
+
+## Adding a New Tool
```

Comment:
It might be worth talking about the relationship between the CLI vs MCP command and how they relate.

Reply:
I don't see a compelling reason (yet) that 99% of new commands we add can't be 1:1 in the cli or mcp.

I'll add some more detail on what you suggest, but I think that once the command structure and response schema has been laid out, most developers should not have to think too deeply about the caller of their tool (cli vs. mcp).